### ISSUE TITLE
ONNX Slice of dynamic tensors

### ISSUE THREAD

**Issue by Liujingxiu23** [pytorch/pytorch#64732]
Export ONNX model

In my model, the input is dynamic, input=[1, seq_len], 1 is batch_size in inference.

After lots of operations, a tensor "X"  got in the process of inference.
And X=[1, 1, seq_len, 10]

After "X" is computed, the following operations are：
x1 = x[:,:,:, 0]
x2 = x[:,:,:, -1]
x3 = x[:,:,:, 1:]
x4 = x[:,:,:,:-1]

But the dynamic dim, dim=2, did not acts as dynamic in x1, x2, x3, x4.

How can I relize the dynamic of these tensors?

cc @BowenBao @neginraoof

**Comment by vigneshwaran** [comment]
I do have similar issue https://github.com/onnx/onnx/issues/3458#issue-871009921

**Comment by jiafatom** [comment]
@Liujingxiu23 Could you provide a small model to illustrate this? It is a little too sketchy here.

**Comment by Liujingxiu23** [comment]
@jiafatom  My model is a little complicated,  funciton "rational_quadratic_spline" in https://github.com/jaywalnut310/vits/blob/2e561ba58618d021b5b8323d3765880f7e0ecfdb/transforms.py#L12.

Maybe you can see this issus in onnx for example. He meet the similar problem.
https://github.com/onnx/onnx/issues/3458

The model is：

import torch
from torch import nn
class Add(nn.Module):
    def __init__(self):
        super().__init__()
    def forward(self, y):
        y = torch.softmax(y, -1)
        y[..., 0] += 0.1
        return y

model = Add()
x = torch.randn(1, 5, 2)
torch.onnx.export(m, x, 'a.onnx', opset_version=12, dynamic_axes={'0': [0,1]})

**Comment by vigneshwaran** [comment]
Thanks for mentioning. Can anyone check on this?

**Comment by garymm** [comment]
As I wrote, please open a new issue for the new problem and include code that reproduces the new problem.
